dagql/cachekey.go

```diff
@@ -46,6 +46,36 @@ func CachePerClientObject[A any](
 	return &cacheCfg, nil
 }
 
+// CachePerSession is a CacheKeyFunc that scopes the cache key to the session by mixing in the session ID to the original digest of the operation.
+// It should be used when the operation should be run for each session, but not more than once for a given session.
+func CachePerSession[P Typed, A any](
```

Comment:
Just thought this would be worth having? It wasn't useful until #9682, but with this, there should be a way to scope cache to just a session (so if two parallel sessions are running, that we won't share results).

Then this'll become more useful when we have *true* global function caching.

Reply:
Added this in https://github.com/dagger/dagger/pull/10184 too (just in case you're confused while rebasing)